.github/workflows/release-artifacts.yml

```diff
@@ -0,0 +1,161 @@
+name: Release artifacts
```

Comment:
Love this splitting up of the release specific workflow but in this repo we still manually have to push tags. Would you suggest adding the auto tag feature in the release script and the tag workflow like [this](https://github.com/RasaHQ/rasa-private/blob/main/.github/workflows/tag-release.yml) to this repo as a follow up after this is merged?

Reply:
I was going for inlining what we have in other repositories.

Reply:
So far we have manually pushed tags so I did not want to make any changes there.
We could add similar script here as well. I would do that in separate PR.